api/health_test.go

```diff
@@ -546,6 +546,69 @@ func TestAPI_HealthConnect_Filter(t *testing.T) {
 	require.Len(t, services, 1)
 }
 
+func TestAPI_HealthConnect_Ingress(t *testing.T) {
+	t.Parallel()
+	c, s := makeClient(t)
+	defer s.Stop()
+
+	agent := c.Agent()
+	health := c.Health()
+
+	s.WaitForSerfCheck(t)
+
+	// Make a service with a proxy
+	reg := &AgentServiceRegistration{
+		Name: "foo",
+		Port: 8000,
+	}
+	err := agent.ServiceRegister(reg)
+	require.NoError(t, err)
+	defer agent.ServiceDeregister("foo")
```

Comment:
I'm not sure if there's a reason to do this deferred deregister - every test makes its own agent so it should all get cleaned up regardless. I see a couple of the other tests in here do it too, though, so maybe it's guarding against something?

Reply:
Yeah not sure, seems like it has been there since the beginning! I removed all `defer agent.ServiceDeregister` in this test file, everything seems fine.